recipe-server/client/control/components/action_fields/HeartbeatFields.js

```diff
@@ -1,99 +1,111 @@
 import React, { PropTypes as pt } from 'react';
+import { connect } from 'react-redux';
 
 import { ControlField } from 'control/components/Fields';
+import { selector } from 'control/components/RecipeForm';
+import ActionFields from 'control/components/action_fields/ActionFields';
 
 /**
  * Form fields for the show-heartbeat action.
  */
-export default function HeartbeatFields({ fields = {} }) {
-  return (
-    <div className="arguments-fields">
-      <p className="info">
-        Shows a single message or survey prompt to the user.
-      </p>
-      <ControlField
-        label="Survey ID"
-        name="arguments.surveyId"
-        component="input"
-        type="text"
-      />
-      <ControlField
-        label="Message"
-        name="arguments.message"
-        component="input"
-        type="text"
-      />
-      <ControlField
-        label="Engagement Button Label"
-        name="arguments.engagementButtonLabel"
-        component="input"
-        type="text"
-      />
-      <ControlField
-        label="Thanks Message"
-        name="arguments.thanksMessage"
-        component="input"
-        type="text"
-      />
-      <ControlField
-        label="Post-Answer URL"
-        name="arguments.postAnswerUrl"
-        component="input"
-        type="text"
-      />
+export class HeartbeatFields extends ActionFields {
+  render({ recipeArguments }) {
```

Comment:
This does need a default, but `initialValues` on the form is probably a better place for it.